package NYT.NChunkServer.NProto;

import "yt/yt/server/lib/chunk_server/proto/job_common.proto";

import "yt_proto/yt/client/node_tracker_client/proto/node.proto";

import "yt_proto/yt/core/misc/proto/guid.proto";

////////////////////////////////////////////////////////////////////////////////

// Informs Job Tracker about node status and requests orders.

message TReqHeartbeat
{
    required uint32 node_id = 1;
    required NNodeTrackerClient.NProto.TNodeDescriptor node_descriptor = 2;
    required NNodeTrackerClient.NProto.TNodeResources resource_usage = 3;
    required NNodeTrackerClient.NProto.TNodeResources resource_limits = 4;
    repeated TJobStatus jobs = 5;

    optional int64 sequence_number = 6 [default = -1];

    required NNodeTrackerClient.NProto.TDiskResources disk_resources = 8;

    // COMPAT(gritukan)
    optional bool reports_heartbeats_to_all_peers = 13;

    reserved 11, 12;
}

message TJobStartInfo
{
    required NYT.NProto.TGuid job_id = 1;
    // Resource limits as predicted by the controller.
    required NNodeTrackerClient.NProto.TNodeResources resource_limits = 3;
}

message TJobToAbort
{
    required NYT.NProto.TGuid job_id = 1;
}

message TJobToRemove
{
    required NYT.NProto.TGuid job_id = 1;
}

message TRspHeartbeat
{
    repeated TJobStartInfo jobs_to_start = 1;
    repeated TJobToAbort jobs_to_abort = 15;
    repeated TJobToRemove jobs_to_remove = 3;

    reserved 2, 5, 7, 9, 10, 12, 13, 14;
}

////////////////////////////////////////////////////////////////////////////////

